.github/workflows/tf-update-lock.yaml

```diff
@@ -0,0 +1,92 @@
+name: Update Terraform Locks
+on:
+  workflow_call:
+    secrets:
+      AWS_ROLE_TO_ASSUME:
+        required: true
+      AWS_REGION:
+        required: true
+      TF_BACKEND_S3_BUCKET:
+        required: true
+      TF_BACKEND_S3_KEY:
+        required: true
+      TF_BACKEND_S3_REGION:
+        required: true
+
+jobs:
+  terraform:
+    name: 'Update Terraform Locks'
+    runs-on: ubuntu-latest
+    # Needed to interact with Github's OIDC token
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.2.4
+
+      - name: Close Previous PR
+        id: close_pr
+        continue-on-error: true
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr close auto_terraform_version
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-session-name: ${{ github.sha }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Terraform Init
+        id: init
+        run: |
+          terraform init -no-color -upgrade \
+            -backend-config="bucket=${{ secrets.TF_BACKEND_S3_BUCKET }}" \
+            -backend-config="key=${{ secrets.TF_BACKEND_S3_KEY }}" \
+            -backend-config="region=${{ secrets.TF_BACKEND_S3_REGION }}"
+
+      - name: Commit Version Lockfile
+        id: commit-lockfile
+        continue-on-error: true
+        run: |
+          git config --local user.email "test@github.com"
+          git config --local user.name "Github Workflow Bot"
+          git add ./.terraform.lock.hcl
+          git commit -m "Update terraform lock"
+
+      - name: Push Version Lockfile
+        uses: ad-m/github-push-action@master
+        if: steps.commit-lockfile.outcome == 'success'
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: auto_terraform_version
+          force: true
+
+      - name: Create Pull Request
+        id: create_pr
+        uses: actions/github-script@v6
+        if: steps.commit-lockfile.outcome == 'success'
+        with:
+          script: |
+            const { repo, owner } = context.repo;
+            const result = await github.rest.pulls.create({
+              title: '[AUTO] Upgrade Terraform Providers to latest',
```

Comment:
This is a bit misleading, in that providers may not be upgrade to latest depending on pin. `Update terraform providers` is more accurate.